packages/tenant-process/src/routers/TenantRouter.ts

```diff
@@ -83,7 +123,25 @@ const tenantsRouter = (
         INTERNAL_ROLE,
         SUPPORT_ROLE,
       ]),
-      async (_req, res) => res.status(501).send()
+      async (req, res) => {
+        try {
+          const tenant = await readModelService.getTenantBySelfcareId(
+            req.params.selfcareId
+          );
+
+          if (tenant) {
+            return res.status(200).json(tenantToApiTenant(tenant.data)).end();
+          } else {
+            return res
+              .status(404)
+              .json(makeApiError(tenantNotFound(req.params.selfcareId)))
```

Comment:
In this case it would be better to have another error named `SelcareIdNotFound`, the error itself should be "Tenant with selfcare id ${selfcareId.toString} not found in the catalog".

Otherwise the `selfcareId` would be confused as the `id` of a `Tenant`